### ISSUE TITLE
`*args` and `**kwargs` types in `call_method`

### ISSUE THREAD

**Issue by jpd-defra** [Defra-Data-Science-Centre-of-Excellence/sds-data-model#229]
When developing the pipeline test, I was getting `mypy` errors in the test code where I was joining two dataframes with the `call_method` method. 

I expected the following code to be using `*args`  rather than `**kwargs` (from `test_pipeline.py`:

```
joined_dfw = (
    spatial_dfw.call_method("join", other=aspatial_dfw.data, on="category")
    .call_method("filter", "land_cover != 'farmland'")
    .categorize(["land_cover"])
    .index()
)
```

I found that adding `Union[SparkDataFrame, GroupedData]` to the *args types in `dataframe.py` did not fix the problem. However, adding `str` and `Union[SparkDataFrame, GroupedData]` to the **kwargs types in `dataframe.py` did fix it. 

To summarise, I think it's worth working out why the call to `call_method` is using non-dict like arguments as `**kwargs.`
